### PR TITLE
Fixes non async call async_forward_entry_setup

### PR DESCRIPTION
**Pull request by rrooggiieerr**
Fixes https://github.com/mrk-its/homeassistant-blitzortung/issues/110

### CODE REVIEW

File: custom_components/blitzortung/__init__.py [hunk 1]
```diff
@@ -89,12 +89,7 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
     hass.data[DOMAIN][config_entry.entry_id] = coordinator
 
     async def start_platforms():
```
Comment (bieniu) [inline custom_components/blitzortung/__init__.py]:
```suggestion
```

File: custom_components/blitzortung/__init__.py [hunk 2]
```diff
@@ -89,12 +89,7 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
...
-        await asyncio.gather(
-            *[
-                hass.config_entries.async_forward_entry_setup(config_entry, component)
-                for component in PLATFORMS
-            ]
-        )
+        await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
```
Comment (bieniu) [inline custom_components/blitzortung/__init__.py]:
```suggestion
    await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
```

File: custom_components/blitzortung/__init__.py [hunk 3]
```diff
@@ -89,12 +89,7 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
...
         await coordinator.connect()
```
Comment (bieniu) [inline custom_components/blitzortung/__init__.py]:
```suggestion
    await coordinator.connect()
```

File: custom_components/blitzortung/__init__.py [hunk 4]
```diff
@@ -89,12 +89,7 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
...
 
     hass.async_create_task(start_platforms())
```
Comment (bieniu) [inline custom_components/blitzortung/__init__.py]:
```suggestion
```

**Review by bieniu — COMMENTED** [review]
`start_platforms` function is not needed.

**Comment by rrooggiieerr** [comment]
@MelleD, nope, not from my side. This pull request has never been accepted by the code owner so I've lost interest. I'm running my own copy with the proposed change and that's working fine

**Comment by bieniu** [comment]
@rrooggiieerr Your PR was not accepted because it was not completely correct (see my comments). If you are not interested in continuing your work, please close it.

**Comment by rrooggiieerr** [comment]
@bieniu, your comment came more than 7 weeks after my pull request, I've never heard anything from the project owner in that time. Issues are also not answered by the project owner. The project seems to abandoned

**Comment by rrooggiieerr** [comment]
@bieniu, actually, I find it a but unfair to ask me to close this pull request as it does fix the issue. Your addition may make it a bit more efficient, but it's not that my pull request contains errors or breaks the code. My pull request could just be accepted and then you could create a pull request which improved further upon that

**Comment by bieniu** [comment]
But you know that you need four clicks to approve my suggestions and make your PR compliant with the current HA code quality standards?

Honestly, I don't care at all whether your or my PR will be merged. I care that the problem is solved in a proper way.

**Comment by rrooggiieerr** [comment]
No, I first need to test your changes on my system before I can assure it works. From first sight it seems like it might work, but I don't think it's good habit to approve untested code. I currently don't have time to dive into it

**Comment by MelleD** [comment]
@rrooggiieerr if you have your own instance and fork you don‘t have to test it. Just apply the suggested changes. I will test it with the next release.

Please all together make this PR not overcomplicated. I also would not merge PRs when standards are not correct.

**Comment by bieniu** [comment]
@rrooggiieerr Yes, you should test the code, but maybe this will give me credibility as someone who knows what he's talking about so you'll feel more comfortable with my suggestions, this is a list of my work in the HA core repo https://github.com/home-assistant/core/pulls?q=is%3Apr+author%3A%40me+is%3Aclosed

**Comment by MelleD** [comment]
> @rrooggiieerr Yes, you should test the code, but maybe this will give me credibility as someone who knows what he's talking about so you'll feel more comfortable with my suggestions, this is a list of my work in the HA core repo https://github.com/home-assistant/core/pulls?q=is%3Apr+author%3Abieniu+is%3Aclosed

Because you have so much experience, I'm so free (or shameless ;) ).

I have a similar custom integration for speed traps and wanted to ask if you could take a look at it in your free time and give feedback. I think you will find a few things with your experience and will definitely help me as a Java developer :), because this PY and HA API is really new for me

It would be great if you gave me a quick answer then I can delete the off-topic post here

**Comment by bieniu** [comment]
@MelleD You can reach me on Discord or Twitter (`bieniu79` on both). I'll try to help.

**Review by bieniu — APPROVED** [review]
Looks good and works as intended 👍

**Comment by rrooggiieerr** [comment]
How about other instances of `hacs.async_create_task`, do these need to be addressed? They're in `geo_location.py` and `mqtt.py`

**Comment by bieniu** [comment]
> How about other instances of `hacs.async_create_task`

Using `async_create_task` itself is absolutely not a problem. In this case the problem was that we were creating a task by `async_create_task` in which we were calling `async_forward_entry_setups` and `async_forward_entry_setups` creates `eager_task` to call `_async_forward_entry_setup`

https://github.com/home-assistant/core/blob/6ef1dd56f582e1887af11ecf95adf812e5ef1a5b/homeassistant/config_entries.py#L2065-L2118

https://github.com/home-assistant/core/blob/6ef1dd56f582e1887af11ecf95adf812e5ef1a5b/homeassistant/util/async_.py#L25

so from a performance point of view this has no sense.

**Comment by rrooggiieerr** [comment]
@bieniu, sorry, I messed up this pull request by committing a new codechange to my main branch which added it automatically to this pull request (I hate git for this). I rolled back my latest change and you need to approve this pull request again.

**Review by bieniu — APPROVED** [review]

**Comment by mrk-its** [comment]
Thanks guys, good job. And sorry it took so long.

**Pull request merged**